sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala

```diff
@@ -49,9 +50,10 @@ abstract class CompactibleFileStreamLog[T <: AnyRef : ClassTag](
 
   import CompactibleFileStreamLog._
 
-  private implicit val formats = Serialization.formats(NoTypeHints)
+  private implicit val formats: Formats = Serialization.formats(NoTypeHints)
 
   /** Needed to serialize type T into JSON when using Jackson */
+  @nowarn
```

Comment:
```suggestion
  @scala.annotation.nowarn
```

Reply:
When this annotation is only needed in one place in this class, it is customary to use xx directly.

Reply:
thanks, get it firstly.